scipy/optimize/_highs/cython/src/_highs_wrapper.pyx

```diff
@@ -653,6 +654,7 @@ def _highs_wrapper(
 
     # We might need an info object if we can look up the solution and a place to put solution
     cdef HighsInfo info = highs.getHighsInfo() # it should always be safe to get the info object
+    cdef HighsRanging highsRanging
```

Comment:
All of these are stack allocated and copied into; would be nice to create a constant ref (doesn't work in Cython if I recall correctly).  Maybe some ptr juggling would be good for these `HighsRanging`, `HighsSolution`, and `HighsBasis` structs